### PR TITLE
Add OSF extension

### PR DESCRIPTION
**Pull request by mih**

### CODE REVIEW

File: .github/workflows/test-datalad_osf.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    - name: WTF!?
+      run: |
+        datalad wtf
+    - name: datalad_osf tests
+      run: |
+        mkdir -p __testhome__
+        cd __testhome__
+
+        python -m nose -s -v --with-cov --cover-package datalad datalad_osf
```
Comment (yarikoptic) [inline .github/workflows/test-datalad_osf.yaml:62]: BTW target package for coverage is datalad, I guess we need to fix template.

File: .github/workflows/test-datalad_ukbiobank.yaml [hunk 2]
```diff
@@ -5,12 +5,11 @@ name: test-datalad_ukbiobank
 
 on:
   # Trigger the workflow on pull request,
-  # but only for the master branch
+  # but only if any given extension workflow was modified
   pull_request:
-    branches:
-      - master
     paths:
```
Comment (yarikoptic) [inline .github/workflows/test-datalad_ukbiobank.yaml:10]: Unrelated to this pr

**Comment by mih** [comment]
It is not clear to me, how to best proceed here. Adding the credentials to this repo is one thing, but should I adjust the generated workflows to make it work? It is not clear to me, whether regeneration is desired? If so, substantial customizations would need to be done. The original repo has functional workflows already, only the datalad installation should differ from what is needed here.

**Comment by yarikoptic** [comment]
I have missed this one? and started #24 which goes one step further

**Pull request closed without merging**